### ISSUE TITLE
BTC-e implementation (offering PR in near future, half-finished)

### ISSUE THREAD

**Issue by juicetin** [butor/blackbird#238]
Hi,

I've currently got a near-finished implementation of BTC-e (in a private repo at the moment), but have some minor issues I haven't been able to sort out I could do with an extra pair of eyes on - will push up my WIP to a public fork under my account if anyone's up for taking a look at what's left. Would be happy to put through a PR once it's working.

I've added `getLimitPrice` and touched up on `getAvail`, both of which seems to work without any problems now

The master branch currently has the `sendLongOrder` function for BTC-e already defined - is it actually complete at the moment? After implementing the above two functions, running Bitfinex/BTC-e gets up to the point where an ENTRY point is found - but the program seg faults before the long order is sent through.  I ran it through gdb to try and pinpoint the issue, no luck - the best I got from it was that the fault occured at `??`, with no library/file/etc. stated at all.

I tried adding extra logging instead, except the failure (probably due to my own error) seems to be occurring right before the long order call from `main.cpp`, but before the function is entered.

**TL;DR** - I've implemented `getLimitPrice` for BTC-e (in a private repo) which seems to be working, but may have issues (unsure if it's the actual/only issue) with `sendLongOrder` which has an existing implementation. If anyone/an existing contributor wants to take a look as well, I'll push up my WIP to a public fork of the repo.

**Comment by greatwolf** [comment]
I just pushed a commit implementing 'getLimitPrice' for BTC-e [also](https://github.com/butor/blackbird/commit/6edb35242c2e34fd723072e49ca8d1a30ca40d72).

I'm not sure why yours is crashing though not without inspecting the code.

**Comment by alfonsoperez** [comment]
Btc-e might not see the light of day again: https://www.justice.gov/usao-ndca/pr/russian-national-and-bitcoin-exchange-charged-21-count-indictment-operating-alleged

**Comment by greatwolf** [comment]
Looks like another US gov shakedown probably going to steal crypto assets from btc-e like they did with silkroad.

**Comment by albabosh** [comment]
There is no BTC-E anymore. It is operated at wex.nx since 15.09
